src/views/ajax.js

```diff
@@ -0,0 +1,112 @@
+const util = require('../lib/util');
+const dom = require('../lib/dom');
+
+let count = 0;
+let ready = false;
+let preRenderCache = [];
+let currentTimout = undefined;
+const summaryStack = [];
```

Comment:
Can this be refactored, to make it easier to use in the future (with or without a framework):

```
const state = {
  count: 0,
  ready: false,
  preRenderCache: [],
  currentTimeout: undefined,
  summaryStack: []
}
```

Motivation: `count` looks like a mysterious global value, but `state.count` at least gives us a clue to where that value comes from, and how we can manage it better in the future.